include/onnxruntime/core/session/onnxruntime_cxx_api.h

```diff
@@ -423,13 +425,27 @@ struct RunOptions : detail::Base<OrtRunOptions> {
 struct SessionOptions;
 
 namespace detail {
+// we separate const-only methods because passing const ptr to non-const methods
+// is only discovered when inline methods are compiled which is counter-intuitive
 template <typename T>
-struct SessionOptionsImpl : Base<T> {
+struct ConstSessionOptionsImpl : Base<T> {
   using B = Base<T>;
   using B::B;
 
   Ort::SessionOptions Clone() const;  ///< Creates and returns a copy of this SessionOptions object. Wraps OrtApi::CloneSessionOptions
 
+  std::string GetConfigEntry(const char* config_key) const;  ///< Wraps OrtApi::GetSessionConfigEntry
+  bool HasConfigEntry(const char* config_key) const;         ///< Wraps OrtApi::HasSessionConfigEntry
+  std::string GetConfigEntryOrDefault(const char* config_key, const std::string& def);
+
+  static const char* const custom_op_config_entry_prefix;
```

Comment:
>static const char* const custom_op_config_entry_prefix; [](http://example.com/codeflow?start=2&length=55)

There might be issues with it. Typically, we need to define an instance in a .cc file, and our CXX api does not have one.

Reply:
Replaced this with a static function that returns a custom op config entry key